sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -728,4 +728,120 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
+          }.toList
+        }
+      }
+
+      def getTableSize: Long = {
+        var totalSize = 0L
+        withTempDir { tmpDir =>
+          withTable(tableName) {
+            createTable(tmpDir)
+            insertOverwriteTable()
+            val path = s"${tmpDir.getPath.stripSuffix("/")}/$tableName"
+            val dir = new File(path)
+            val files = getDirFiles(dir).filter(_.getName.startsWith("part-"))
+            totalSize = files.map(_.length()).sum
+          }
+        }
+        totalSize
+      }
+    }
+
+    def checkParquetCompressionCodec(isPartitioned: Boolean, tableCodec: String,
+      sessionCodec: String, f: (Long, Long) => Boolean = _ == _): Unit = {
+      val tableOrg = TableDefine(s"tbl_parquet$tableCodec", isPartitioned, "parquet",
+        Some(CompressionConf("parquet.compression", tableCodec)))
+      val tableOrgSize = tableOrg.getTableSize
+
+      withSQLConf("spark.sql.parquet.compression.codec" -> sessionCodec) {
+        // priority check, when table-level compression conf was set, expecting
+        // table-level compression conf is not affected by the session conf, and table-level
+        // compression conf takes precedence even the two conf of codec is different
+        val tableOrgSessionConfSize = tableOrg.getTableSize
+        assert(tableOrgSize == tableOrgSessionConfSize)
+
+        // check session conf of compression codec taking effect
+        val table = TableDefine(s"tbl_parquet", isPartitioned, "parquet", None)
+        assert(f(tableOrg.getTableSize, table.getTableSize))
+      }
+    }
+
+    def checkOrcCompressionCodec(isPartitioned: Boolean, tableCodec: String,
+      sessionCodec: String, f: (Long, Long) => Boolean = _ == _): Unit = {
+      val tableOrg = TableDefine(s"tbl_orc$tableCodec", isPartitioned, "orc",
+        Some(CompressionConf("orc.compress", tableCodec)))
+      val tableOrgSize = tableOrg.getTableSize
+
+      withSQLConf("spark.sql.orc.compression.codec" -> sessionCodec) {
+        // priority check, when table-level compression conf was set, expecting
+        // table-level compression conf is not affected by the session conf, and table-level
+        // compression conf takes precedence even the two conf of codec is different
+        val tableOrgSessionConfSize = tableOrg.getTableSize
+        assert(tableOrgSize == tableOrgSessionConfSize)
+
+        // check session conf of compression codec taking effect
+        val table = TableDefine(s"tbl_orc", isPartitioned, "orc", None)
+        assert(f(tableOrg.getTableSize, table.getTableSize))
+      }
+    }
+
+    withTempView("table_source") {
+      (0 until 100000).toDF("a").createOrReplaceTempView("table_source")
+
+      checkParquetCompressionCodec(true, "UNCOMPRESSED", "UNCOMPRESSED")
+      checkParquetCompressionCodec(true, "GZIP", "GZIP")
+      checkParquetCompressionCodec(true, "GZIP", "UNCOMPRESSED", _ < _)
+
+      withSQLConf("spark.sql.hive.convertMetastoreParquet" -> "false") {
```

Comment:
We need to prove this PR works in both cases globally.
- **true** and **false** of `convertMetastoreParquet` for all combinations (line 828~835)
- **true** and **false** of `convertMetastoreOrc` for all combinations (line 838~844)

If you use the real compression kinds instead of size, it will be easy.

Reply:
If this is due to the issue of partitioned tables, could you add comments here?